tests/integration/bigquery.spec.ts

```diff
@@ -144,6 +163,70 @@ describe("@dataform/integration/bigquery", () => {
     }
   });
 
+  describe("publish tasks", async () => {
+    it("incremental, core version <= 1.4.8", async () => {
+      const projectConfig: dataform.IProjectConfig = {
+        warehouse: "bigquery",
+        defaultDatabase: "default_database"
+      };
+
+      const table: dataform.ITable = {
+        type: "incremental",
+        query: "query",
+        preOps: ["preop task1", "preop task2"],
+        incrementalQuery: "query where incremental",
+        postOps: ["postop task1", "postop task2"],
+        target: {
+          schema: "df_integration_test",
+          name: "example_incremental",
+          database: "dataform-integration-tests"
+        }
+      };
+
+      const expectedRefreshStatements = [
+        table.preOps[0],
+        table.preOps[1],
+        `drop view if exists \`${table.target.database}.${table.target.schema}.${table.target.name}\``,
+        `create or replace table \`${table.target.database}.${table.target.schema}.${table.target.name}\`  as ${table.query}`,
+        table.postOps[0],
+        table.postOps[1]
+      ];
+
+      const expectedIncrementStatements = [
+        table.preOps[0],
+        table.preOps[1],
+        `drop view if exists \`${table.target.database}.${table.target.schema}.${table.target.name}\``,
+        `
+insert into \`${table.target.database}.${table.target.schema}.${table.target.name}\`
+()
+select 
+from (
+  select * from (${table.incrementalQuery}) as subquery
+    where true) as insertions`,
+        table.postOps[0],
+        table.postOps[1]
+      ];
+
+      const bqadapter = new BigQueryAdapter(projectConfig, "1.4.8");
+
+      const buildsFromRefresh = bqadapter
+        .publishTasks(table, { fullRefresh: true }, { fields: [] })
+        .build();
+
+      buildsFromRefresh.forEach((build, i) => {
+        expect(build.statement).to.eql(expectedRefreshStatements[i]);
+      });
+
+      const buildsFromIncrement = bqadapter
+        .publishTasks(table, { fullRefresh: false }, { fields: [] })
+        .build();
+
+      buildsFromIncrement.forEach((build, i) => {
+        expect(build.statement).to.eql(expectedIncrementStatements[i]);
+      });
+    });
+  });
+
```

Comment:
I think this is as short as I can get it, and doesn't read too bad IMO. Refresh and increment statements do differ between different adapters.